File: src/lua/types/lttorrentinfo.cpp
#include "../types.hpp"

#include <libtorrent/load_torrent.hpp>
#include <libtorrent/torrent_info.hpp>

using porla::Lua::Types::LtTorrentInfo;

void LtTorrentInfo::Register(sol::state& lua)
{
    lua.new_usertype<lt::torrent_info>(
        "LtTorrentInfo",
        sol::no_constructor,
        "info_hash",   &lt::torrent_info::info_hashes,
        "layout",      &lt::torrent_info::layout,
        "name",        &lt::torrent_info::name,
        "num_files",   &lt::torrent_info::num_files,
        "num_pieces",  &lt::torrent_info::num_pieces,
        "priv",        &lt::torrent_info::priv,
        "total_size",  &lt::torrent_info::total_size);
}

File: types/lua/usertypes/lttorrentinfo.lua
---@meta
---@class LtTorrentInfo
local LtTorrentInfo = {}

---@return LtInfoHash
function LtTorrentInfo:info_hash() end

---@return LtFileStorage
function LtTorrentInfo:layout() end

---@return string
function LtTorrentInfo:name() end

---@return integer
function LtTorrentInfo:num_files() end

---@return integer
function LtTorrentInfo:num_pieces() end

---@return boolean
function LtTorrentInfo:priv() end

---@return integer
function LtTorrentInfo:total_size() end
